bothan-coingecko/src/api/builder.rs

```diff
@@ -32,32 +32,50 @@ pub struct CoinGeckoRestAPIBuilder {
 }
 
 impl CoinGeckoRestAPIBuilder {
+    pub fn new<T, U, V>(url: Option<T>, api_key: Option<U>, user_agent: V) -> Self
```

Comment:
- I think optional parameters should come after the required argument.
- I think if we want to take `Option` we should not use type param (just use `Option<String>`) otherwise when passing `None` user will need to supply type explicitly which is quite frustrating.